src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -90,35 +91,25 @@
 {
     private static final Logger logger = LoggerFactory.getLogger(RepairRunnable.class);
 
+    private static final AtomicInteger THREAD_COUNTER = new AtomicInteger(1);
+    private static final int TOTAL_PROGRESS = CoordinatorState.State.NUM_STATES;
+
+    public final CoordinatorState state;
     private final StorageService storageService;
-    private final int cmd;
-    private final RepairOption options;
-    private final String keyspace;
 
     private final String tag;
-    private final AtomicInteger progressCounter = new AtomicInteger();
-    private final int totalProgress;
-
-    private final long creationTimeMillis = currentTimeMillis();
-    private final UUID parentSession = UUIDGen.getTimeUUID();
 
     private final List<ProgressListener> listeners = new ArrayList<>();
-
-    private static final AtomicInteger threadCounter = new AtomicInteger(1);
     private final AtomicReference<Throwable> firstError = new AtomicReference<>(null);
 
     private TraceState traceState;
 
     public RepairRunnable(StorageService storageService, int cmd, RepairOption options, String keyspace)
     {
+        this.state = new CoordinatorState(cmd, keyspace, options);
         this.storageService = storageService;
-        this.cmd = cmd;
-        this.options = options;
-        this.keyspace = keyspace;
 
         this.tag = "repair:" + cmd;
-        // get valid column families, calculate neighbors, validation, prepare for repair + number of ranges to repair
-        this.totalProgress = 4 + options.getRanges().size();
```

Comment:
I am doing a breaking change here as total progress is kinda meaningless in repair as we expose a % and this acts kinda like a step counter, so we go from 5/100 to 6/100, to 100/100